common/snapshots.py

```diff
@@ -698,6 +698,15 @@ def remove(self, sid):
 
             return True
 
+    def _warning_on_take_snapshot(self, config):
```

Comment:
This function name (`_warning_on_take_snapshot`) does indicate that a warning is issued but it is unclear under which condition (or what is checked).

I suggest to rename the function to eg. `_check_included_sources_exist_on_take_snapshot` (perhaps you find a better word than "sources")